### DISCUSSION TITLE
How to disable caching within the request class itself?

### DISCUSSION THREAD

**Discussion by boryn** [saloonphp/saloon#187] [Q&A]
I know there is possbility to disable cache on a per-request basis (like `$request->disableCaching();`).

But I'm looking for a way to disable it within the request class. I tried to put `$this->disableCaching();` in an overridden `boot()` method, but it did not help. 

Why? Actually I may cache results on the development environment only, and cache cannot be used in production. That's why I'm looking for something like:

```php
if (App::environment('local')) {
    $this->enableCaching();
} else {
    $this->disableCaching();
}
```

**Comment by Sammyjo20** [comment]
Hey @boryn the `boot()` method on a connector/request runs after the plugins have already booted, so this is probably why the enable/disable caching methods didn't work there for you - have you tried putting this logic in your __construct of your connector/request?

**Comment by boryn** [comment]
Hi @Sammyjo20!

Yes, putting `$this->disableCaching()` in the constructor helped and now I can disable it on the fly:

```php
public function __construct()
{
    $this->disableCaching();
}
```

On the other hand I believe disabling caching should automatically invoke clearing the cache. Now, let's have this scenario:
1. enable cache
2. do the request query
3. it gets stored in the cache
4. run the request again
5. it gets fetched from the cache - great
6. disable cache
7. run the request again
8. it gets fetched through the "fresh" call
9. enable cache
10. it gets fetched from the **"old" cache**, stored in the step 3

IMHO in the step 7 when the cache got disabled we should remove the cache record, so that when we run step 9 and 10, it should get response from the fresh call and save it to the cache (instead of servicing a stale cache record).

Additionaly, setting this:
```php
public function __construct()
{
    $this->disableCaching();

    $this->invalidateCache();
}
```

does not remove cache record at all, so either it's a bug, or I run it in a wrong order, but at this moment I have no idea how to clear the existing cache record :) Using `return new LaravelCacheDriver(Cache::store('database'));`
